include/lsst/ctrl/events/CommandEvent.h

```diff
@@ -63,38 +61,38 @@ namespace events {
 class CommandEvent : public Event
 {
 public:
-    static const std::string ORIGINATORID;
-    static const std::string ORIG_LOCALID;
+    static const std::string ORIG_HOSTNAME;
     static const std::string ORIG_PROCESSID;
-    static const std::string ORIG_IPID;
+    static const std::string ORIG_LOCALID;
 
-    static const std::string DESTINATIONID;
-    static const std::string DEST_LOCALID;
+    static const std::string DEST_HOSTNAME;
     static const std::string DEST_PROCESSID;
-    static const std::string DEST_IPID;
+    static const std::string DEST_LOCALID;
 
     CommandEvent();
-    CommandEvent(const std::string& runid, const int64_t originatorId, const int64_t destinationId, const PropertySet& ps);
-    CommandEvent(const std::string& runid, const int64_t originatorId, const int64_t destinationId, const PropertySet::Ptr psp);
+
+    CommandEvent(const LocationID& originator, const LocationID& destination, const PropertySet::Ptr psp);
+    CommandEvent(const LocationID& originator, const LocationID& destination, const PropertySet& ps);
+    CommandEvent(const LocationID& originator, const LocationID& destination, const PropertySet& ps, const PropertySet& filterable);
+
+    CommandEvent(const std::string& runid, const LocationID& originator, const LocationID& destination, const PropertySet::Ptr psp);
+    CommandEvent(const std::string& runid, const LocationID& originator, const LocationID& destination, const PropertySet& ps);
+    CommandEvent(const std::string& runid, const LocationID& originator, const LocationID& destination, const PropertySet& ps, const PropertySet& filterable);
+
     CommandEvent(cms::TextMessage *msg);
-    virtual void populateHeader(cms::TextMessage *msg) const;
 
     virtual ~CommandEvent();
 
-    int64_t getOriginatorId();
-    short getOriginatorLocalId();
-    int  getOriginatorProcessId();
-    int getOriginatorIPId();
+    LocationID *getOriginator();
```

Comment:
And shouldn't you return a reference or const reference instead of a raw pointer?